python/lsst/cell_coadds/_fits.py

```diff
@@ -69,17 +125,71 @@ class CellCoaddFitsReader:
         The name of the FITS file to read.
     """
 
+    # Minimum and maximum compatible file format versions are listed as
+    # iterables so as to allow for discontiguous intervals.
+    MINIMUM_FILE_FORMAT_VERSIONS = ("0.1",)
+    MAXIMUM_FILE_FORMAT_VERSIONS = ("1.0",)
+
     def __init__(self, filename: str) -> None:
         if not os.path.exists(filename):
             raise FileNotFoundError(f"File {filename} not found")
 
         self.filename = filename
 
+    @classmethod
+    def isCompatibleWith(cls, written_version: str, /) -> bool:
+        """Check if the serialization version is compatible with the reader.
+
+        Parameters
+        ----------
+        written_version: `str`
+            The version to compare against.
+
+        Returns
+        -------
+        compatible : `bool`
+            Whether the reader is compatible with ``written_version``.
+
+        Notes
+        -----
+        This accepts the other version as a positional argument only.
+        """
+        for min_version, max_version in zip(
+            cls.MINIMUM_FILE_FORMAT_VERSIONS,
+            cls.MAXIMUM_FILE_FORMAT_VERSIONS,
+            strict=True,
+        ):
+            if min_version <= written_version < max_version:
```

Comment:
I didn't realize this kind of construct actually worked in Python.  Are you sure it does?

Reply:
Yes, it does! Here's some random blog from 12 years ago: https://www.johndcook.com/blog/2012/04/03/a-b-c/

And by `a <= b < c`, I do mean `a < b` and `b < c` which is exactly how Python interprets it as well.

Reply:
:exploding_head:

Reply:
Oh, I should also note that I do mean to compare strings here, not after casting them to `float`.